### DISCUSSION TITLE
Stacked CLI with Mock Services

### DISCUSSION THREAD

**Discussion by SeanningTatum** [FilledStacks/stacked#908] [Q&A] [answered]
Hi - I'm a big fan of using "abstraction / mock services" as it really helps with e2e testing and mocking things before creating an API. But I'm struggling to do this in the new stacked cli setup.

```dart title="app.locator.dart"
Future<void> setupLocator(
    {String? environment, EnvironmentFilter? environmentFilter}) async {
// Register environments
  locator.registerEnvironment(
      environment: environment, environmentFilter: environmentFilter);

// Register dependencies
  locator.registerLazySingleton(() => BottomSheetService());
  locator.registerLazySingleton(() => DialogService());
  locator.registerLazySingleton(() => NavigationService());
  locator.registerLazySingleton(
    () => environment == Environment.dev
        ? CounterMockService()
        : CounterRealService(),
  );
}
```
So based on the old video https://www.youtube.com/watch?v=DO8le1W_HqQ, this is generally how we'd switch from using a MockService and a RealService. 

I was wondering how we would currently set this up with the new cli, as the generators would delete this, and the test generators go crazy.

**Comment by FilledStacks** [comment, marked as answer]
@SeanningTatum good question, currently the CLI doesn't support that. But you can do that setup through the `StackedApp` by doing something like this. 

```dart
LazySingleton(
  asType: BaseRouterService,
  classType: RouterService,
  environments: {Environment.prod},
),
LazySingleton(
  asType: BaseRouterService,
  classType: MockRouterService,
  environments: {Environment.dev, Environment.test},
),
```

Check if that gives you what you're hoping for.

**Reply by SeanningTatum** [reply]
Yes this does answer my question! and a quick note to anyone else who reads this thread - after editing the `StackedApp` as above, run `stacked generate` to see the autogenerated services in `app.locator.dart`
